### PR TITLE
Remove ossec.log file from the macOS package building

### PR DESCRIPTION
**Pull request by verdx**
|Related issue|
|---|
| #2209  |

## Description
This PR solves a problem with building macOS packages. If, during the building of the package, something got onto the log file, `/Library/Ossec/logs/ossec.log` before the package was finally created, it would pass on to every system into which that package is installed, as explained in #2209. This is solved by removing the log file before completely creating the package.

## Tests

The test has consisted of forcing a write on the log file inside the building script to check if it gets to the package installation, which it doesn't.

The following line has been added to `build.sh`:
```
    echo "Testing" >> /Library/Ossec/logs/ossec.log
```

The package is then created: https://ci.wazuh.info/job/Packages_builder_macos/9923/

And when the package is installed, the log file is empty:

```
sh-3.2# launchctl setenv WAZUH_MANAGER "10.0.0.2" && installer -pkg wazuh-agent-4.4.3-2209\(1\).pkg -target / 
installer: Package name is Wazuh Agent
installer: Installing at base path /
installer: The install was successful.
sh-3.2#  cat /Library/Ossec/logs/ossec.log
```

Checking no other files are installed
```
sh-3.2# ls /Library/Ossec
.ssh		active-response	agentless	bin		etc		lib		logs		queue		ruleset		tmp		var		wodles
sh-3.2# ls /Library/Ossec/logs
active-responses.log	ossec.json		ossec.log		wazuh
sh-3.2# ls /Library/Ossec/logs/wazuh/
```

### CODE REVIEW

File: rpms/SPECS/wazuh-manager.spec [hunk 1]
```diff
@@ -842,6 +842,8 @@ rm -fr %{buildroot}
 %attr(750, root, wazuh) %{_localstatedir}/wodles/gcloud/*
 
 %changelog
+* Mon May 22 2023 support <info@wazuh.com> - 4.4.3
```
Comment (rauldpm) [inline rpms/SPECS/wazuh-manager.spec:845]: Wrong date

**Review by rauldpm — CHANGES_REQUESTED** [review]

**Comment by verdx** [comment]
# 4.4 to 4.4.3 changes in GitHub
In the `Files changed` tab in GitHub it appears as if the changes from the bump from 4.4 to 4.4.3 are included in this PR. This comes from doing a rebase mid-development of the PR. These changes are not really included in the branch, which can be seen doing a `git diff`:

```
diff --git a/macos/package_files/build.sh b/macos/package_files/build.sh
index f94537ef..8f9f388a 100755
--- a/macos/package_files/build.sh
+++ b/macos/package_files/build.sh
@@ -76,6 +76,8 @@ function build() {
 
     cp ${SOURCES_PATH}/src/VERSION ${INSTALLATION_SCRIPTS_DIR}/src/
     cp ${SOURCES_PATH}/src/REVISION ${INSTALLATION_SCRIPTS_DIR}/src/
+
+    rm /Library/Ossec/logs/ossec.log
 }
 
 build
diff --git a/macos/uninstall.sh b/macos/uninstall.sh
index 169ed9fc..6f894e64 100755
--- a/macos/uninstall.sh
+++ b/macos/uninstall.sh
@@ -1,7 +1,7 @@
 #/bin/sh
 
 ## Stop and remove application
-sudo /Library/Ossec/bin/ossec-control stop
+sudo /Library/Ossec/bin/wazuh-control stop
 sudo /bin/rm -r /Library/Ossec*
 
 ## stop and unload dispatcher
```

For this reason I think the requested changes by @rauldpm should not be done in this PR, but in a different Issue and PR, done only to fix the errors in the bump.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,9 @@
 # Change Log
 All notable changes to this project will be documented in this file.
+## [v4.4.3]
```
[the remark this answers is not in the document]
Comment (verdx) [reply CHANGELOG.md:3]: Why not to do this change here explained in https://github.com/wazuh/wazuh-packages/pull/2216#issuecomment-1560637572

File: rpms/SPECS/wazuh-manager.spec (hunk 1 above)
Comment (verdx) [reply rpms/SPECS/wazuh-manager.spec:845]: Why not to do this change here explained in https://github.com/wazuh/wazuh-packages/pull/2216#issuecomment-1560637572

**Review by DFolchA — APPROVED** [review]

**Review by rauldpm — CHANGES_REQUESTED** [review]
We cannot accept the proposed solution since we are not fixing the root problem but applying a palliative to the problem.

The ossec.log file should not be part of the final package, so we have no guarantee that there may be other new/changed files that shouldn't be in the package.

Currently, this ossec.log file is the only way we have to know that the package is not acceptable, so we should investigate the root of the problem and apply a solution at a lower level.

**Review by DFolchA — CHANGES_REQUESTED** [review]

**Comment by verdx** [comment]
# Update: closing this PR

After talking with the rest of the team, it has been decided this PR be closed. If the problem happens with this file, it could happen with others, but `ossec.log` would be written for sure, as it reflects the agent service has started without a correct manager ip. That way, only removing this file just prevents us from knowing if the agent service has been started during the building of the package.

The presence of logs in these files should be checked when the packages are created, and if they are written, a new package should be created. For the next release, `4.4.4`, the error will continue to be looked for in more detail in the issue https://github.com/wazuh/wazuh-packages/issues/2209.

The other fix present in this PR, for the binary in the uninstall script, has been transferred to PR https://github.com/wazuh/wazuh-packages/pull/2207.

**Pull request closed without merging**